### PR TITLE
Goals: Implement goals in Report Budget

### PR DESCRIPTION
**Pull request by shall0pass**
This will enable goals in the Report Budget.

### CODE REVIEW

File: packages/desktop-client/src/components/budget/index.js [hunk 1]
```diff
@@ -516,6 +516,16 @@ const RolloverBudgetSummary = memo(props => {
   );
 });
 
+const ReportBudgetSummary = memo(props => {
+  const isGoalTemplatesEnabled = useFeatureFlag('goalTemplatesEnabled');
```
Comment (j-f1) [inline packages/desktop-client/src/components/budget/index.js:520]: Since `BudgetSummary` is a function component, couldn’t this hook call be moved into there to avoid the wrapping component?

File: packages/loot-core/src/server/budget/goaltemplates.ts [hunk 2]
```diff
@@ -356,7 +357,7 @@ async function applyCategoryTemplate(
         } else {
           increment = limit;
         }
-        if (to_budget + increment < budgetAvailable || !priority) {
+        if (increment < budgetAvailable || !priority) {
```
Comment (j-f1) [inline packages/loot-core/src/server/budget/goaltemplates.ts:360]: Was this change intentional?

File: packages/loot-core/src/server/budget/goaltemplates.ts [hunk 3]
```diff
@@ -563,10 +569,15 @@ async function applyCategoryTemplate(
           amountCond.value = monthlyTarget;
         }
 
-        if (template.full === true) {
-          if (num_months === 1) {
+        if (template.full === true || isReflectBudget()) {
+          if (num_months === 0) {
```
Comment (j-f1) [inline packages/loot-core/src/server/budget/goaltemplates.ts:573]: ^ same here — was this change intentional?

File: packages/loot-core/src/server/budget/goaltemplates.ts (hunk 2 above)
Comment (shall0pass) [reply packages/loot-core/src/server/budget/goaltemplates.ts:360]: I think this was in the PR that was just merged.  I made the change to test it.  I can revert it back in this PR.

File: packages/loot-core/src/server/budget/goaltemplates.ts (hunk 3 above)
Comment (shall0pass) [reply packages/loot-core/src/server/budget/goaltemplates.ts:573]: This was intentional.  The calculation wasn't happening.  I think it broke when we started using the monthUtils package to calculate dates.

File: packages/desktop-client/src/components/budget/index.js (hunk 1 above)
Comment (shall0pass) [reply packages/desktop-client/src/components/budget/index.js:520]: The reason I did it this was because I was copying the method used in the above function since it's solving the same problem for the Rollover budget.  I can try to implement what you're asking though.

**Comment by shall0pass** [comment]
I saw that I inadvertently changed a file I didn't intend to.  Troubles with working on two files named BudgetSummary.tsx

File: packages/desktop-client/src/components/budget/report/BudgetSummary.tsx [hunk 4]
```diff
@@ -301,6 +303,8 @@ export const BudgetSummary = memo(function BudgetSummary({
     onToggleSummaryCollapse,
   } = useReport();
 
+  isGoalTemplatesEnabled = useFeatureFlag('goalTemplatesEnabled');
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/budget/report/BudgetSummary.tsx:306]: **❓ question:** ‏is this line necessary? To me it look like we already get the value on line 297.

Comment (shall0pass) [reply packages/desktop-client/src/components/budget/report/BudgetSummary.tsx:306]: It doesn't work without it.  I'll change this line to ` const isGoalTemplatesEnabled = useFeatureFlag('goalTemplatesEnabled');` and remove lines 293 and 297.  That appears to work for me.

Comment (MatissJanis) [reply packages/desktop-client/src/components/budget/report/BudgetSummary.tsx]: Cool, I'm ok with that.

I'm happy to approve the PR if you can solve the merge conflict.

**Comment by migillett** [comment]
so hyped for this

**Review by j-f1 — APPROVED** [review]
